src/create_disk.sh

```diff
@@ -364,7 +383,16 @@ install_grub_cfg() {
     # 0700 to match the RPM permissions which I think are mainly in case someone has
     # manually set a grub password
     mkdir -p -m 0700 $rootfs/boot/grub2
-    printf "%s\n" "$grub_script" > $rootfs/boot/grub2/grub.cfg
+    printf "%s\n" "$grub_script" | \
+        sed -E 's@(^# CONSOLE-SETTINGS-START$)@\1'"${platform_grub_cmds:+\\n${platform_grub_cmds}}"'@' \
+        > $rootfs/boot/grub2/grub.cfg
+    if [ -n "$platforms_json" ]; then
+        # Copy platforms table if it's non-empty for this arch
+        if jq -e ".$arch" < "$platforms_json" > /dev/null; then
+            mkdir -p "$rootfs/boot/coreos"
+            jq ".$arch" < "$platforms_json" > "$rootfs/boot/coreos/platforms.json"
```

Comment:
Remind me again, why does the coreos-installer side require it in `/boot`? If it were only in `/usr`, could it not just mount the rootfs to find it? I know I initially was the one in favour of [only having it in `/boot`](https://github.com/coreos/fedora-coreos-config/pull/1181#issuecomment-905596476), but now with https://github.com/coreos/coreos-assembler/pull/2839, there is a push for shipping "image-building metadata" in the rootfs.

So, could we ship this not in `/boot` but instead in `/usr/lib/coreos-assembler/` alongside `image.json`? You said in another comment to not block on this, but not sure if you feel the same if a change in path was in the picture. Otherwise, I fear we'll indeed end up with it both in `/boot` and `/usr` (which obviously isn't the end of the world, but it'd be nice to have a single source of truth).

Reply:
The installer already mounts `/boot` and doesn't currently know or care about `/`.  We could change that, but the installer would also need to navigate the ostree repo, tying `install -p` to ostree.  To me that seems like a worse tradeoff than maintaining a private copy of `platforms.json` for use by cosa.

(Even if we moved `platforms.json` to `/usr`, it shouldn't go in `/usr/lib/coreos-assembler`, since that path strongly implies that it's a private internal API.  Maybe something like `/usr/share/coreos/platforms.json`?)

Note that the path change wouldn't be sufficient for #2839 compatibility, since we're currently injecting `platforms.json` during image build and not during ostree build.

Reply:
Yeah, tricky. Seems doable still, but unsure about the tradeoff also.